.github/pull_request_template.md

```diff
@@ -1,4 +1,22 @@
+## Why
+
+<!-- Explain why the changes are needed.  -->
+
 Fixes #
 
-Changes proposed in this pull request:
+## What
+
+<!-- Describe changes proposed in this pull request. -->
+
+## Tests
+
+<!-- Describe how the change was tested (both manual and automated) for reviewers to find edge cases more easily. -->
+
+## Checklist
+
+<!-- All items should be verified and marked as done.
+     ~~strikethrough~~ if an item doesn't apply to the introduced changes. -->
 
+- [ ] `CHANGELOG.md` is updated.
+- [ ] Documentation is updated.
+- [ ] New features are covered by tests.
```

Comment:
If this step is skipped when a feature is added, I think we should call out why it is skipped. For example, with the metrics work, in order to make the PRs easier to review the work could be split up in such a way to add the test framework/harness code in a separate PR.

Reply:
Yeah, the way I see it listing those explicitly invite consideration about the items, but, we can always accept a justification to not cover those.

Reply:
1. I think we should avoid such cases. I think that adding "new signals support" like metrics and logs is unique and we do not have to call it out in this template.
2. If we skip this step then we should call out (in the PR description) why it is skipped and create a GitHub issue for tracking.

Reply:
@pellared In general I agree with you. While we are still in these early stages and we are still improving our testing strategy we might have more of these larger testing gaps. Do we want to have the exception process you are recommending defined in the template, or do we think it's sufficient for approvers and maintainers to enforce it without including more info the template? I'm ok with either approach.

Reply:
I agree with @pellared bullets 1 and 2. I would rather not add anything about possible exceptions on the template itself. We want processes and checklists to invite/guide good practices, but, we don't want to dogmatically and blindingly be constrained by them - one needs to be careful to not make processes/checklists useless by having too many exceptions.  Achieving that requires a bit of judgement/trade-offs but I think our SIG is up to the task :)